lib/core/src/persist/chain.rs

```diff
@@ -98,13 +99,21 @@ impl Persister {
     }
 
     pub(crate) fn insert_or_update_chain_swap(&self, chain_swap: &ChainSwap) -> Result<()> {
+        let maybe_swap = self.fetch_chain_swap_by_id(&chain_swap.id)?;
+        let updated_fields = ChainSyncData::updated_fields(maybe_swap, chain_swap);
+
         let mut con = self.get_connection()?;
         let tx = con.transaction_with_behavior(TransactionBehavior::Immediate)?;
 
         Self::insert_or_update_chain_swap_inner(&tx, chain_swap)?;
-        self.commit_outgoing(&tx, &chain_swap.id, RecordType::Chain, None)?;
-        tx.commit()?;
-        self.sync_trigger.try_send(())?;
+        // Trigger a sync if:
+        // - updated_fields is None (swap is inserted, not updated)
+        // - updated_fields in a non empty list of updated fields
+        if updated_fields.as_ref().map_or(true, |u| !u.is_empty()) {
+            self.commit_outgoing(&tx, &chain_swap.id, RecordType::Chain, updated_fields)?;
+            tx.commit()?;
+            self.sync_trigger.try_send(())?;
+        }
```

Comment:
Just making sure we are not missing a commit in case we don't need to sync. Is it implicit on transaction drop?

Reply:
Oof, yeah. That's bad, we should always commit

Reply:
Fixed in f866be6526d1